Neos.RedirectHandler.DatabaseStorage/Classes/Domain/Model/Redirection.php

```diff
@@ -0,0 +1,239 @@
+<?php
+namespace Neos\RedirectHandler\DatabaseStorage\Domain\Model;
+
+/*
+ * This file is part of the Neos.RedirectHandler package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Doctrine\ORM\Mapping as ORM;
+use TYPO3\Flow\Annotations as Flow;
+use Gedmo\Mapping\Annotation as Gedmo;
+
+/**
+ * A Redirection model that represents a HTTP redirect
+ *
+ * @see RedirectionService
+ *
+ * @Flow\Entity
+ * @ORM\Table(
+ * 	indexes={
+ * 		@ORM\Index(name="targeturipathhash",columns={"targeturipathhash","hostpattern"})
+ * 	}
+ * )
+ */
+class Redirection
+{
+    /**
+     * @var \DateTime
+     */
+    protected $creationDateTime;
+
+    /**
+     * @var \DateTime
+     * @Gedmo\Timestampable(on="update", field={"sourceUriPath", "targetUriPath", "statusCode", "hostPattern"})
+     */
+    protected $lastModificationDateTime;
+
+    /**
+     * Auto-incrementing version of this node data, used for optimistic locking
+     *
+     * @ORM\Version
+     * @var integer
+     */
+    protected $version;
+
+    /**
+     * Relative URI path for which this redirect should be triggered
+     *
+     * @var string
+     * @ORM\Column(length=4000)
+     */
+    protected $sourceUriPath;
+
+    /**
+     * MD5 hash of the Source Uri Path
+     *
+     * @var string
+     * @ORM\Column(length=32)
+     * @Flow\Identity
+     */
+    protected $sourceUriPathHash;
+
+    /**
+     * Target URI path to which a redirect should be pointed
+     *
+     * @var string
+     * @ORM\Column(length=500)
+     */
+    protected $targetUriPath;
+
+    /**
+     * MD5 hash of the Target Uri Path
+     *
+     * @var string
+     * @ORM\Column(length=32)
+     */
+    protected $targetUriPathHash;
+
+    /**
+     * Status code to be send with the redirect header
+     *
+     * @var integer
+     * @Flow\Validate(type="NumberRange", options={ "minimum"=100, "maximum"=599 })
+     */
+    protected $statusCode;
+
+    /**
+     * Host Pattern
```

Comment:
That pattern is not a pattern… Since 2013, when we added that HostnameValidator there. See this commit in my pipeline: https://github.com/kdambekalns/neos-development-collection/commit/79465245aab2561630febcf7f72e37c558f3f9ed